example/ipss_app/parse_and_plot.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python
+"""
+Simple (non automatic) validation script of the inhomgeneous Poisson Spike Source
+cell.
+"""
+import matplotlib.pyplot as plt
+import collections
+import math
+import sys
+
+default_spike_file_path = "./spikes.gdf"
+multiplier = 30.0
+
+def usage():
+    print """
+Usage: 
+python parse_and_plot.py [path_spikes] [path_time_rate] [plot_target interpolate]
+
+Arguments:
+ -path_spikes    (../spikes)path to gdf formatted file with spikes
+ -path_time_rate file path with float, float pairs per line with time rate pairs
+ -plot_target    Plot the (expected) target rates as a line
+ -interpolate    Interpolate between the time-rate pairs
+
+parse_and_plot.py is a simple script to validate the performance of the 
+ipss cell. No input validation is performed.
+
+- Parse gdf spike file
+- Performs a binning operation for the first 1000 ms 
+  - with 1 ms binning 
+  - averages the found rates assuming a total of 10000 cells.
+- Displays a figure with the histogram of the instantatious spike rate for
+the whole group, including a target rate.
+
+Default target rate: 
+ hz.|                     
+ 240|     _-_             
+    |    -   -  -         
+    |   -     -- -        
+ 0  |__-__________-__     
+      100        900   ms 
+
+Todo:
+ - Add statistical test for correct output for automated validation
+"""
+
+default_time_rate_pairs = [[0.0   , 0.0 * multiplier],
+                           [50.0  , 0.0 * multiplier],
+                           [100.0 , 1.0 * multiplier],
+                           [200.0 , 5.0 * multiplier],
+                           [300.0 , 7.0 * multiplier],
+                           [400.0 , 8.0 * multiplier],
+                           [500.0 , 7.0 * multiplier],
+                           [600.0 , 3.0 * multiplier],
+                           [700.0 , 3.0 * multiplier],
+                           [750.0 , 5.0 * multiplier],
+                           [800.0 , 2.5 * multiplier],
+                           [900.0 , 0.0 * multiplier],
+                           [1000.0, 0.0 * multiplier]]
+
+def parse_spike_file(path):
+    """ 
+    Simple spike file parsing function. Assumes no errors and might fail 
+    silently on errors
+    """
+    spikes_per_cell = collections.defaultdict(list)
+    with open(path , "r") as f:
+        for line in f:
+            tokens = line.split(",")
+            try:
+                gid = int(tokens[0].strip())
+                time = float(tokens[1].strip())
+            except:
```

Comment:
Is there only one possible exception here? Couldn't the exception tell us something other than failed?